src/main/java/org/broadinstitute/consent/http/resources/DataSetResource.java

```diff
@@ -57,16 +60,42 @@
     private final String END_OF_LINE = System.lineSeparator();
     private final String TSV_DELIMITER = "\t";
     private final DataSetAPI api;
+    private final DatasetService datasetService;
     private final DataAccessRequestAPI dataAccessRequestAPI;
     private final UserService userService;
 
     @Inject
-    public DataSetResource(UserService userService) {
+    public DataSetResource(DatasetService datasetService, UserService userService) {
         this.dataAccessRequestAPI = AbstractDataAccessRequestAPI.getInstance();
         this.api = AbstractDataSetAPI.getInstance();
+        this.datasetService = datasetService;
         this.userService = userService;
     }
 
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("/test")
+    @PermitAll
+    public Response createDataset(@Auth AuthUser user, String json) {
+        DataSet ds = new Gson().fromJson(json, DataSet.class);
```

Comment:
I think we need to do something a little different here. Since datasets have raw dataset properties, that means your incoming json string has to mimic that structure exactly. This leads to an unfriendly situation where the client needs to know what the property key integer values are. What I'd like to see instead is input that looks almost exactly like the output (minus the things that the system calculates):
```
{
  "active": true, 
  "consentId": "DEV-CG-4412", 
  "dacId": 4, 
  "deletable": null, 
  "isAssociatedToDataOwners": null, 
  "needsApproval": false, 
  "objectId": null, 
  "properties": [
    {
      "propertyName": "Dataset Name", 
      "propertyValue": "TOPMed CCAF_GRU_IRB_phs001189.v1.p1.c1"
    }, 
    {
      "propertyName": "Data Type", 
      "propertyValue": "Whole Genome"
    }, 
    {
      "propertyName": "Species", 
      "propertyValue": "human"
    }, 
    {
      "propertyName": "Phenotype/Indication", 
      "propertyValue": "Atrial Fibrilation"
    }, 
    {
      "propertyName": "# of participants", 
      "propertyValue": "357"
    }, 
    {
      "propertyName": "Description", 
      "propertyValue": "Cleveland Clinic Atrial Fibrillation Study"
    }, 
    {
      "propertyName": "dbGAP", 
      "propertyValue": "https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs001189.v1.p1"
    }, 
    {
      "propertyName": "Data Depositor", 
      "propertyValue": "NHLBI"
    }, 
    {
      "propertyName": "Principal Investigator(PI)", 
      "propertyValue": "NHLBI"
    }
  ], 
  "translatedUseRestriction": "Samples are restricted for use under the following conditions:\nData is available for general research use. [GRU]\nCommercial use is not prohibited.\nData use for methods development research irrespective of the specified data use limitations is not prohibited.\nFuture use as a control set for any type of health/medical/biomedical study is not prohibited.\nOther restrictions: IRB Approval Required.", 
  "updateAssociationToDataOwnerAllowed": null
}
```
I think if you deserialize as a `DataSetDTO`, then you could backport that object to a dataset and then save it.